x-pack/protocol/src/main/java/org/elasticsearch/protocol/xpack/rollup/RollupField.java

```diff
@@ -1,9 +1,22 @@
 /*
- * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
- * or more contributor license agreements. Licensed under the Elastic License;
- * you may not use this file except in compliance with the Elastic License.
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
  */
-package org.elasticsearch.xpack.core.rollup;
+package org.elasticsearch.protocol.xpack.rollup;
```

Comment:
Note: I moved RollupField along with the rollup configuration objects as it is used in their toBuilders() and getMetadata()  methods. I created #32585 and #32579 to remove these two methods from the config objects but we didn't reach a consensus on it (see @polyfractal https://github.com/elastic/elasticsearch/pull/32585#issuecomment-410033057) so any opinion on this is welcome. I don't think it's blocking the PR either.

Reply:
Still mulling this over, but after seeing how it integrates in HLRC (namely, being moved to `protocol` package) I'm thinking it may be better to separate from the config object despite my earlier objections.  

The HLRC will require yet another package to keep updated (`xpack.rollup`, `xpack.core`, `xpack.protocol` and the HLRC itself).  So probably best to just move `toBuilders()` and `getMetadata()` into the rollup package.  The logic will still be spread out, but won't require hunting across multiple packages at least.

Thoughts? 

I agree it's not blocking though